recipes/tesseract/all/conanfile.py

```diff
@@ -92,13 +112,21 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
         # remove cmake
         tools.rmdir(os.path.join(self.package_folder, 'cmake'))
+        # required for 5.0
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["pthread"]
-        elif self.settings.compiler == "Visual Studio":
-            if not self.options.shared:
-                self.cpp_info.system_libs = ["ws2_32"]
         self.cpp_info.names["cmake_find_package"] = "Tesseract"
         self.cpp_info.names["cmake_find_package_multi"] = "Tesseract"
+
+        self.cpp_info.components["libtesseract"].libs = tools.collect_libs(self)
+        self.cpp_info.components["libtesseract"].requires = ["leptonica::leptonica", "libarchive::libarchive" ]
+
+        self.cpp_info.components["libtesseract"].names["cmake_find_package"] = "libtesseract"
+        self.cpp_info.components["libtesseract"].names["cmake_find_package_multi"] = "libtesseract"
+        self.cpp_info.components["libtesseract"].names["pkg_config"] = "libtesseract"
+
+        if self.settings.os == "Linux":
+            self.cpp_info.components["libtesseract"].system_libs = ["pthread"]
+        elif self.settings.compiler == "Visual Studio" and not self.options.shared:
+            self.cpp_info.components["libtesseract"].system_libs = ["ws2_32"]
```

Comment:
Not sure we should hind the dependency

```suggestion
        elif self.settings.compiler == "Visual Studio":
            self.cpp_info.components["libtesseract"].system_libs = ["ws2_32"]
```

Reply:
Agreed. Checked other recipes, they usually link `ws2_32` even on shared.